nesta_ds_utils/loading_saving/S3.py

```diff
@@ -46,9 +47,15 @@ def _df_to_fileobj(df_data: pd.DataFrame, path_to: str, **kwargs) -> io.BytesIO:
         df_data.to_csv(buffer, **kwargs)
     elif fnmatch(path_to, "*.parquet"):
         df_data.to_parquet(buffer, **kwargs)
+    elif fnmatch(path_to, "*.xlsx"):
+        df_data.to_excel(buffer, **kwargs)
+    elif fnmatch(path_to, "*.xlsm"):
+        df_data.to_excel(buffer, **kwargs)
+    elif fnmatch(path_to, "*.geojson"):
+        df_data.to_file(buffer, driver="GeoJSON", **kwargs)
     else:
         raise Exception(
-            "Uploading dataframe currently supported only for 'csv' and 'parquet'."
+            "Uploading dataframe currently supported only for 'csv', 'parquet', 'xlsx', xlsm' and 'geojson'."
```

Comment:
may be worth specifying this as a separate function given it's a geopandas df rather than a pandas df?

Reply:
I don't mind either way! I think a geopandas df still counts as a pandas df. The only difference being an extra geometry column that has a particular typing and that the package itself has extra functionality for geojsons etc. But happy to separate into another function, if that makes readability better?

Reply:
I think maybe? On the geojson question, I've never needed it but someone might! So will add it in!

Reply:
Latest geopandas docs indicates that geopandas df is a separate object type even though the library is supposedly built on top of pandas.

I think you'd need to add a instance type check condition for `gpd.GeoDataFrame` for each `fnmatch` branch, and raise an appropriate error message whether the filetype is implemented for GeoDataFrames.

Linking below some reference to supported IO operations for GeoPandas:
- [User Guide](https://geopandas.org/en/stable/docs/user_guide/io.html#)
- [IO methods](https://geopandas.org/en/stable/docs/reference/io.html)
- [GeoPandas instance methods](https://geopandas.org/en/stable/docs/reference/geodataframe.html#)
- Note, there is some overlap between the IO methods and "instance methods" sections, but there's a few extra `to_something` methods in the latter.

Reply:
Ah fab! So the isinstance(geodf, pd.DataFrame ) returns true for geopandas dataframe, but it doesnt work in the other direction. So it must count as pandas somewhere in it. I'll just make a separate function!

Reply:
I'm slightly confused, did you do instance checking on a gpd.GeoDataFrame or pd.DataFrame?

Reply:
You could also try:
```
from geopandas.base import GeoPandasBase
```

issubclass(gpd.GeoDataFrame, GeoPandasBase)

Reply:
Ooo I like that! Thanks so much for checking everything :)